services/api-server/src/simcore_service_api_server/utils/client_base.py

```diff
@@ -30,36 +34,82 @@ async def is_responsive(self) -> bool:
             resp.raise_for_status()
             return True
         except (httpx.HTTPStatusError, httpx.RequestError) as err:
-            log.error("%s not responsive: %s", self.service_name, err)
+            _logger.error("%s not responsive: %s", self.service_name, err)
             return False
 
     ping = is_responsive  # alias
 
 
+class _AsyncClientWithCaptures(httpx.AsyncClient):
+    """
+    Adds captures mechanism
+    """
+
+    async def request(self, method: str, url: URLTypes, **kwargs):
+        response: httpx.Response = await super().request(method, url, **kwargs)
+
+        capture_name = f"{method} {url}"
+        _logger.info("Capturing %s ... [might be slow]", capture_name)
+        try:
+            capture_json = get_capture_msg(name=capture_name, response=response)
+            _capture_logger.info("%s", capture_json)
+        except ValidationError:
+            _capture_logger.exception("Failed capturing %s", capture_name)
+
+        return response
+
+
 # HELPERS -------------------------------------------------------------
 
+_capture_logger = logging.getLogger(f"{__name__}.capture")
+
+
+def _setup_capture_logger_once(capture_path: Path) -> None:
+    """NOTE: this is only to capture during developmetn"""
+
+    if not any(
+        isinstance(hnd, logging.FileHandler) for hnd in _capture_logger.handlers
+    ):
+        file_handler = logging.FileHandler(filename=f"{capture_path}")
+        file_handler.setLevel(logging.INFO)
+
+        formatter = logging.Formatter("%(asctime)s - %(message)s")
+        file_handler.setFormatter(formatter)
+
+        _capture_logger.addHandler(file_handler)
+        _logger.info("Setup capture logger at %s", capture_path)
+        _capture_logger.info("Started capture session ...")
+
 
 def setup_client_instance(
     app: FastAPI,
     api_cls: type[BaseServiceClientApi],
     api_baseurl,
     service_name: str,
-    **extra_fields
+    **extra_fields,
 ) -> None:
     """Helper to add init/cleanup of ServiceClientApi instances in the app lifespam"""
 
     assert issubclass(api_cls, BaseServiceClientApi)  # nosec
 
+    # Http client class
+    client_class: type = httpx.AsyncClient
+    with suppress(AttributeError):  # State not having settings
+        if capture_path := app.state.settings.API_SERVER_HTTP_CALLS_CAPTURE_LOGS_PATH:
```

Comment:
I would have expected `API_SERVER_HTTP_CALLS_CAPTURE_LOGS_PATH` to be None not to raise an AttributeError.

Reply:
The attribute error is expected when the ``state.settings``  is undefined, not the ``API_SERVER_HTTP_CALLS_CAPTURE_LOGS_PATH`` 

That is what I mean in the comment. I will clarify it